misk/src/main/kotlin/misk/web/actions/WebProxyAction.kt

```diff
@@ -0,0 +1,108 @@
+package misk.web.actions
+
+import misk.scope.ActionScoped
+import misk.security.authz.Unauthenticated
+import misk.web.Get
+import misk.web.Post
+import misk.web.Request
+import misk.web.RequestContentType
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.ResponseContentType
+import misk.web.mediatype.MediaTypes
+import misk.web.mediatype.asMediaType
+import misk.web.toMisk
+import misk.web.toResponseBody
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import java.io.IOException
+import java.net.HttpURLConnection
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+
+/**
```

Comment:
Can we bundle all of the proxy stuff up into a dedicated package? It's difficult to differentiate now what is required for systems using proxies support, and what isn't. Given that proxies imply required configuration (http clients and such) it would be good to have it all separately located vs intermingled with API action support